test/common/http/http1/codec_impl_test.cc

```diff
@@ -142,6 +142,14 @@ TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePath2) {
   expectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
 }
 
+TEST_F(Http1ServerConnectionImplTest, Http11AbsolutePathWithPort) {
+  TestHeaderMapImpl expected_headers{
+      {":authority", "www.somewhere.com:4532"}, {":path", "/foo/bar"}, {":method", "GET"}};
+  Buffer::OwnedImpl buffer(
+      "GET http://www.somewhere.com:4532/foo/bar HTTP/1.1\r\nHost: bah\r\n\r\n");
+  expectHeadersTest(Protocol::Http11, true, buffer, expected_headers);
+}
```

Comment:
Though looking more at the code, if that doesn't work end to end it's more likely a problem with exiting route picking than this particular PR.   So I'd still encourage testing and either fixing here or in a follow-up since ports are much more common with absolute URIs and this fix won't do much for you if you can't pick routes :-)

My only other comment is it'd be nice to have regression tests for the bit-shifting fix if it's doable    But the code here looks totally solid.